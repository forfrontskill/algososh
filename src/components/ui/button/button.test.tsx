import React from 'react';
import { Button } from './button';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
describe('Test Button component', () => {
    
    it('Button rendering', () => {
        const tree = renderer
          .create(<Button text='ButtonTextTest' />)
          .toJSON();
          expect(tree).toMatchSnapshot();
      });

    it('Button creation', () => {
        const context = render(<Button text='ButtonTextTest' />);
    
        const button = screen.getByRole('button');
    
        expect(button).toBeInTheDocument();
    });

    it('Button with text', () => {
        const testText = 'ButtonTextTest';
        const context = render(<Button text={testText} />);

        const buttonText = screen.getByTestId('button-text').textContent;
    
        expect(buttonText).toBe(testText);
    });

    it('Button without text', () => {
        const testText = '';
        const context = render(<Button disabled/>);

        const buttonText = screen.getByTestId('button-text').textContent;
    
        expect(buttonText).toBe(testText);
    });

    it('Button disabled', () => {
        const testText = '';
        const context = render(<Button disabled/>);

        const button = screen.getByRole('button');
    
        expect(button).toBeDisabled();
    });

    it('Button is loading', () => {
        const context = render(<Button isLoader={true}/>);

        const button = screen.getByRole('button');
        const loader = screen.getByAltText('Загрузка.');
        expect(button).toBeInTheDocument();
        expect(loader).toBeInTheDocument();
    });
})
