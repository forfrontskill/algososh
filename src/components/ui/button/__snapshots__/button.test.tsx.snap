// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Button rendering 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    className="text undefined"
    data-testid="button-text"
  >
    ButtonTextTest
  </p>
</button>
`;
